Add DevOps Projects (#6772)
kamranahmedse/developer-roadmap: src/data/projects/basic-github-actions.md

---
title: 'Github Actions Workflow'
description: 'Write  GitHub Actions workflow to deploy a simple GitHub Pages site.'
isNew: true
sort: 3
difficulty: 'beginner'
nature: 'CI/CD'
skills:
  - 'devops'
  - 'github actions'
  - 'ci/cd'
seo:
  title: 'Github Actions Workflow'
  description: 'Write GitHub Actions workflow to deploy a simple GitHub Pages site.'
  keywords:
    - 'basic ci/cd'
    - 'devops'
    - 'devops projects'
roadmapIds:
  - 'git-github'
  - 'devops'
---

In this project, you will write a basic HTML file and setup a GitHub Actions workflow to test, build & deploy it to GitHub Pages.

## Requirements

- Create a simple HTML file (the content is up to you)
- Create a GitHub Actions workflow that will test, build & deploy the website to [GitHub Pages](https://pages.github.com/).
- Failures in the workflow should be clearly indicated and failures will halt the workflow

> - The workflow should be in the `.github/workflows` directory
> - The workflow file should be named `main.yml`
> 
> 
> <hr />

Author: ```suggestion
- The workflow should be in the `.github/workflows` directory
- The workflow file should be named `main.yml`

<hr />
```

---
title: 'Github Actions Workflow'

> description: 'Write a basic GitHub Actions workflow to deploy an Astro GitHub Pages site.'

Author: ```suggestion
description: 'Write  GitHub Actions workflow to deploy an simple GitHub Pages site.'
```

isNew: true
sort: 3
difficulty: 'beginner'
nature: 'CI/CD'
skills:
  - 'devops'
  - 'github actions'
  - 'ci/cd'

>   - 'astro'
> seo:
>   title: 'Github Actions Workflow'
>   description: 'Write a basic GitHub Actions workflow to deploy an Astro GitHub Pages site.'
>   keywords:
>     - 'basic ci/cd'
>     - 'devops'
>     - 'devops projects'
> roadmapIds:
>   - 'git-github'
>   - 'devops'
> ---
> 
> In this project, you will build a simple Astro website and setup a GitHub Actions workflow to test, build & deploy it to GitHub Pages.
> 
> ## Requirements
> 
> - Create a simple Astro website (the content of the website is up to you)
> - Create a GitHub Actions workflow that will test, build & deploy the website to [GitHub Pages](https://pages.github.com/).
> - Failures in the workflow should be clearly indicated and failures will halt the workflow
> - The workflow should be in the `.github/workflows` directory
> - The workflow file should be named `main.yml`

Author: ```suggestion
seo:
  title: 'Github Actions Workflow'
  description: 'Write GitHub Actions workflow to deploy a simple GitHub Pages site.'
  keywords:
    - 'basic ci/cd'
    - 'devops'
    - 'devops projects'
roadmapIds:
  - 'git-github'
  - 'devops'
---

In this project, you will write a basic HTML file and setup a GitHub Actions workflow to test, build & deploy it to GitHub Pages.

## Requirements

- Create a simple HTML file (the content is up to you)
- Create a GitHub Actions workflow that will test, build & deploy the website to [GitHub Pages](https://pages.github.com/).
- Failures in the workflow should be clearly indicated and failures will halt the workflow
- The workflow should be in the `.github/workflows` directory
- The workflow file should be named `main.yml`
```

> You can learn more about creating an Astro project [here](https://docs.astro.build/en/getting-started/).

Author: ```suggestion
```
